mita/openstack.py

```diff
@@ -310,63 +310,46 @@ def _wait_until_ip_is_known(self):
-                _vol = self.driver_v2.create_volume(
-                    self.size_of_disk, "{}-vol-{}".format(self.node_name, item)
-                )
-                self.volumes.append(_vol)
-            except BaseException as be:  # noqa
-                logger.debug("Failed to create volume.")
-                logger.debug(be)
+            vol_name = f"{self.node_name}-vol-{item}"
+            logger.info("Creating %s volume with %sGib", vol_name, self.size_of_disk)
+            volume = self.driver_v2.create_volume(self.size_of_disk, vol_name)
+
+            if not volume:
+                raise VolumeOpError(f"Failed to create volume with name {vol_name}")
+
+            self.volumes.append(volume)
 
         for _vol in self.volumes:
-            self._wait_until_volume_available(_vol, maybe_in_use=True)
+            if not self._wait_until_volume_available(_vol):
+                raise VolumeOpError(f"{_vol.name} failed to become available.")
 
         for _vol in self.volumes:
             if not self.driver_v2.attach_volume(self.node, _vol):
-                raise VolumeAttachmentError("Unable to attach volume %s", _vol.name)
+                raise VolumeOpError("Unable to attach volume %s", _vol.name)
 
             logger.info("Successfully attached %s to %s", _vol.name, self.node_name)
 
-    def _wait_until_volume_available(self, volume, maybe_in_use=False):
-        """
-        Wait until a StorageVolume's state is "available".
-        Set "maybe_in_use" to True in order to wait even when the volume is
-        currently in_use. For example, set this option if you're recycling
-        this volume from an old node that you've very recently
-        destroyed.
-        """
-        ok_states = ["creating"]  # it's ok to wait if the volume is in this
+    def _wait_until_volume_available(self, volume: StorageVolume) -> bool:
+        """Wait until the state of the StorageVolume is available."""
         tries = 0
-        if maybe_in_use:
-            ok_states.append("in_use")
-        logger.info("Volume: %s is in state: %s", volume.name, volume.state)
-
-        while volume.state in ok_states:
+        logger.debug("%s waiting for volume to become available.", volume.name)
+        while True:
             sleep(3)
-            volume = self.get_volume(volume.name)
-            tries = tries + 1
-            if tries > 10:
-                logger.info("Maximum amount of tries reached..")
+            tries += 1
+            volume = self.driver_v2.ex_get_volume(volume.id)
+
+            if volume.state.lower() == "available":
+                return True
+
+            if "error" in volume.state.lower():
+                logger.error("%s state is %s", volume.name, volume.state)
                 break
-            if volume.state == "notfound":
-                logger.error("no volume was found for: %s", volume.name)
+
+            if tries > 10:
+                logger.error("Maximum amount of retries reached.")
```

Comment:
Let's log the volume name here, `logger.error(f'Max retries for {volume.name} reached')`